prov/rxd/src/rxd_init.c

```diff
@@ -119,6 +121,10 @@ static int rxd_getinfo(uint32_t version, const char *node, const char *service,
 			uint64_t flags, const struct fi_info *hints,
 			struct fi_info **info)
 {
+	if (rxd_env.always_rescan)
+		flags |= FI_RESCAN;
+	else
+		flags &= ~FI_RESCAN;
```

Comment:
If the user doesn't set any environment variable, this has the impact of removing the FI_RESCAN flag from their request.  That's probably not what we want.  I would leave the flags unchanged in this case.

Reply:
I updated the code and `always_rescan` is now an integer that can have the following values:
- `-1`: The user didn't set the environment variable
- `0`: rescan explicitly disabled by the environment variable
- `1`: rescan explicitly enabled by the environment variable

Note that the value is now retrieved with `fi_param_get_bool()` but the underlying value is effectively an integer (and not a boolean).

Reply:
Maybe convert to an enum?  It should be okay to use get_bool, as that will leave the value alone if the variable isn't defined.  I'd rename to just 'rescan' if false means never.

It seems that this would be a common flow (unset, false, true), but I don't see anything similar in tree based on a quick search.

Reply:
I renamed the parameter to just 'rescan' as you suggested but I kept the boolean. Converting it to an enum will require some extra work and I think it is beyond the scope of this PR.